packages/webapp/src/containers/SensorReadings/index.jsx

```diff
@@ -1,78 +1,52 @@
-    nearestStationName = '',
-    lastUpdatedReadingsTime = '',
-    predictedXAxisLabel = '',
-    xAxisLabel = {},
-  } = useSelector(bulkSensorsReadingsSliceSelector);
+  const dispatch = useDispatch();
+  const { location_id } = match?.params;
   const [readingTypes, setReadingTypes] = useState([]);
-  const unitSystem = useSelector(measurementSelector);
+  const [locationData, setLocationData] = useState();
+
+  const sensorInfo = useSelector(sensorsSelector(location_id));
   const reading_types = useSelector(sensorReadingTypesByLocationSelector(location_id));
-  const [sensorVisualizationPropList, setSensorVisualizationPropList] = useState({});
+  const { loading, sensorDataByLocationIds } = useSelector(bulkSensorsReadingsSliceSelector);
 
+  //Keeps sensor readings up to date for location
+  useEffect(() => {
+    if (sensorDataByLocationIds[location_id] && location_id) {
+      setLocationData(sensorDataByLocationIds[location_id]);
+    }
+  }, [sensorDataByLocationIds, location_id]);
+
+  // Handles unknown records and keeping readingTypes up to date
   useEffect(() => {
     if (sensorInfo === undefined || reading_types === undefined || sensorInfo?.deleted) {
       history.replace('/unknown_record');
     } else {
-      const loadedReadingTypes = reading_types.reading_types;
-      setReadingTypes(loadedReadingTypes);
-      setSensorVisualizationPropList({
-        [TEMPERATURE]: {
-          title: t('SENSOR.TEMPERATURE_READINGS_OF_SENSOR.TITLE'),
-          subTitle: t('SENSOR.TEMPERATURE_READINGS_OF_SENSOR.SUBTITLE', {
-            high: latestMaxTemperature,
-            low: latestMinTemperature,
-            units: getUnitOptionMap()[ambientTemperature[unitSystem].defaultUnit].label,
-          }),
-          weatherStationName: t('SENSOR.TEMPERATURE_READINGS_OF_SENSOR.WEATHER_STATION', {
-            weatherStationLocation: nearestStationName,
-          }),
-          yAxisLabel: t('SENSOR.TEMPERATURE_READINGS_OF_SENSOR.Y_AXIS_LABEL', {
-            units: getUnitOptionMap()[ambientTemperature[unitSystem].defaultUnit].label,
-          }),
-          ambientTempFor: t('SENSOR.TEMPERATURE_READINGS_OF_SENSOR.AMBIENT_TEMPERATURE_FOR'),
-        },
-        [SOIL_WATER_POTENTIAL]: {
-          title: t('SENSOR.SOIL_WATER_POTENTIAL_READINGS_OF_SENSOR.TITLE'),
-          yAxisLabel: t('SENSOR.SOIL_WATER_POTENTIAL_READINGS_OF_SENSOR.Y_AXIS_LABEL', {
-            units: getUnitOptionMap()[soilWaterPotential[unitSystem].defaultUnit].label,
-          }),
-        },
-        [SOIL_WATER_CONTENT]: {
-          title: t('SENSOR.SOIL_WATER_CONTENT_READINGS_OF_SENSOR.TITLE'),
-          yAxisLabel: t('SENSOR.SOIL_WATER_CONTENT_READINGS_OF_SENSOR.Y_AXIS_LABEL', {
-            units: '%',
-          }),
-        },
-      });
+      setReadingTypes(reading_types.reading_types);
+    }
+  }, [sensorInfo, reading_types]);
+
+  //Runs the saga update store
+  useEffect(() => {
+    if (location_id && readingTypes.length) {
+      dispatch(
```

Comment:
This triggers re-rendering, but I think this should happen to get the latest data.